SerbianC 2.2 patch for babel source tree (updated from svn 2011/06/06)
Index: babel/babel.dtx
===================================================================
--- babel.orig/babel.dtx	2011-06-06 13:05:30.119057290 +0200
+++ babel/babel.dtx	2011-06-06 13:06:23.355057310 +0200
@@ -333,6 +333,7 @@
 %        Slovenian  & slovene\\
 %        Swedish    & swedish\\
 %        Serbian    & serbian\\
+%	 Serbian cyrillic & serbianc\\
 %^^A        Tamil      & tamil \\
 %        Turkish    & turkish\\
 %        Ukrainian  & ukrainian\\
@@ -416,6 +417,8 @@
 %    \item Support for typesetting Serbian with Latin script is now
 %      available thanks to Dejan Muhamedagi\'{c} and Jankovic
 %      Slobodan.
+%    \item Support for typesetting Serbian with cyrillic script is now
+%      availible thanks to Filip Br\v{c}i\'{c}.
 %    \item Support for typesetting Hebrew (and potential support for
 %      typesetting other right-to-left written languages) is now
 %      available thanks to Rama Porrat and Boris Lavva.
@@ -1168,6 +1171,7 @@
 %^^A\DeclareOption{sanskrit}{\input{sanskrit.ldf}}
 \DeclareOption{scottish}{\input{scottish.ldf}}
 \DeclareOption{serbian}{\input{serbian.ldf}}
+\DeclareOption{serbianc}{\input{serbianc.ldf}}
 \DeclareOption{slovak}{\input{slovak.ldf}}
 \DeclareOption{slovene}{\input{slovene.ldf}}
 \DeclareOption{spanish}{\input{spanish.ldf}}
@@ -5830,6 +5834,7 @@
 \langdeffile{czech.dtx}
 \langdeffile{polish.dtx}
 \langdeffile{serbian.dtx}
+\langdeffile{serbianc.dtx}
 \langdeffile{slovak.dtx}
 \langdeffile{slovene.dtx}
 \langdeffile{russianb.dtx}
Index: babel/babel.ins
===================================================================
--- babel.orig/babel.ins	2011-06-06 13:05:30.147057290 +0200
+++ babel/babel.ins	2011-06-06 13:06:23.359057310 +0200
@@ -217,6 +217,7 @@
 %\BblInstall{sanskrit}{sanskrit}
 \BblInstall{scottish}{scottish}
 \BblInstall{serbian}{serbian}
+\BblInstall{serbianc}{serbianc}
 \BblInstall{slovak}{slovak}
 \BblInstall{slovene}{slovene}
 \BblInstall{sorbian}{sorbian}
Index: babel/base.ins
===================================================================
--- babel.orig/base.ins	2011-06-06 13:05:35.983057292 +0200
+++ babel/base.ins	2011-06-06 13:06:23.359057310 +0200
@@ -250,6 +250,7 @@
           \compatfile{latin}
           \compatfile{icelandic}
           \compatfile{serbian}
+	  \compatfile{serbianc}
           \compatfile{bulgarian}
           }
 \generate{%
Index: babel/serbianc.dtx
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ babel/serbianc.dtx	2011-06-06 13:07:21.775057330 +0200
@@ -0,0 +1,239 @@
+% \iffalse meta-comment
+%
+% Copyright 1989-2001 Johannes L. Braams and any individual authors
+% listed elsewhere in this file.  All rights reserved.
+%
+% This file is part of the Babel system release 3.7.
+% --------------------------------------------------
+%
+% It may be distributed under the terms of the LaTeX Project Public
+% License, as described in lppl.txt in the base LaTeX distribution.
+% Either version 1.2 or, at your option, any later version.
+% \fi
+% \CheckSum{263}
+% \iffalse
+%    Tell the \LaTeX\ system who we are and write an entry on the
+%    transcript.
+%<*dtx>
+\ProvidesFile{serbianc.dtx}
+%</dtx>
+%<code>\ProvidesLanguage{serbianc}
+%\fi
+%\ProvidesFile{serbianc.dtx}
+       [2011/06/06 v2.2 Serbian cyrillic]
+%\iffalse
+% Babel package for LaTeX version 2e
+% Copyright (C) 1989 - 2001
+%           by Johannes Braams, TeXniek
+%
+% Please report errors to: J.L. Braams
+%                          JLBraams@cistron.nl
+%
+%    This file is part of the babel system, it provides the source
+%    code for the Serbian cyrillic language definition file.  A contribution
+%    was made by Filip Br\v{c}i\'{c} <brcha@gna.org>
+%
+%<*filedriver>
+\documentclass{ltxdoc}
+\newcommand*\TeXhax{\TeX hax}
+\newcommand*\babel{\textsf{babel}}
+\newcommand*\langvar{$\langle \it lang \rangle$}
+\newcommand*\note[1]{}
+\newcommand*\Lopt[1]{\textsf{#1}}
+\newcommand*\file[1]{\texttt{#1}}
+\begin{document}
+ \DocInput{serbianc.dtx}
+\end{document}
+%</filedriver>
+%\fi
+% \GetFileInfo{serbianc.dtx}
+% \changes{serbianc-1.0}{2002/09/03}{Created the first public release
+% of serbian cyrillic packet for babel}
+% \changes{serbianc-1.1}{2005/11/09}{Released new version with some bugs fixed
+% and with automated installer for GNU/* or POSIX-like systems}
+% \changes{serbianc-2.0}{2006/11/24}{Dropped support for non-utf coding style,
+% now you have to use T2A font encoding and write using utf-8. If you don't like
+% that, use previous version}
+% \changes{serbianc-2.1}{2007/02/23}{Minor bugfixes}
+% \changes{serbianc-2.2}{2011/06/06}{Published on CTAN}
+%
+%  \section{The Serbian cyrillic language}
+%
+%    The file \file{\filename}\footnote{The file described in this
+%    section has version number \fileversion\ and was last revised on
+%    \filedate.  A contribution was made by Filip Br\v{c}i\'{c}
+%    (\texttt{brcha@gna.org}).}  Defines all the language
+%    definition macros for the serbianc language, typeset in a cyrillic
+%    script.
+%
+%    Apart from defining shorthands we need to make sure that the
+%    first paragraph of each section is indented. Furthermore the
+%    following new math operators are defined (|\tg|, |\ctg|,
+%    |\arctg|, |\arcctg|, |\sh|, |\ch|, |\th|, |\cth|, |\arsh|,
+%    |\arch|, |\arth|, |\arcth|, |\Prob|, |\Expect|, |\Variance|).
+%
+% \StopEventually{}
+%
+%    The macro |\LdfInit| takes care of preventing that this file is
+%    loaded more than once, checking the category code of the
+%    \texttt{@} sign, etc.
+%    \begin{macrocode}
+%<*code>
+\LdfInit{serbianc}\captionsserbianc
+%    \end{macrocode}
+%
+%    When this file is read as an option, i.e. by the |\usepackage|
+%    command, \texttt{serbianc} will be an `unknown' language in which
+%    case we have to make it known. So we check for the existence of
+%    |\l@serbianc| to see whether we have to do something here.
+%
+%    \begin{macrocode}
+\ifx\l@serbianc\@undefined
+    \@nopatterns{serbianc}
+    \adddialect\l@serbianc0\fi
+%    \end{macrocode}
+%
+%    \begin{macrocode}
+
+\AtEndOfPackage{%
+    \RequirePackage{ucs}%
+    \PassOptionsToPackage{utf8x}{inputenc}%
+    \RequirePackage{inputenc}%
+    \PassOptionsToPackage{T2A}{fontenc}%
+    \RequirePackage{fontenc}%
+}
+
+%    \end{macrocode}
+%
+%    The next step consists of defining commands to switch to (and
+%    from) the Serbocroatian language.
+%
+%  \begin{macro}{\captionsserbianc}
+%    The macro |\captionsserbianc| defines all strings used
+%    in the four standard documentclasses provided with \LaTeX.
+%    \begin{macrocode}
+\addto\captionsserbianc{%
+  \def\prefacename{\CYRP\cyrr\cyre\cyrd\cyrg\cyro\cyrv\cyro\cyrr}%{Предговор}%
+  \def\refname{\CYRL\cyri\cyrt\cyre\cyrr\cyra\cyrt\cyru\cyrr\cyra}%{Литература}%
+  \def\abstractname{\CYRA\cyrb\cyrs\cyrt\cyrr\cyra\cyrk\cyrt}%{Абстракт}%
+  \def\bibname{\CYRB\cyri\cyrb\cyrl\cyri\cyro\cyrg\cyrr\cyra\cyrf\cyri\cyrje\cyra}%
+            %{Библиографија}%
+  \def\chaptername{\CYRG\cyrl\cyra\cyrv\cyra}%{Глава}%
+  \def\appendixname{\CYRD\cyro\cyrd\cyra\cyrt\cyra\cyrk}%{Додатак}%
+  \def\contentsname{\CYRS\cyra\cyrd\cyrr\cyrzh\cyra\cyrje}%{Садржај}%
+  \def\listfigurename{\CYRL\cyri\cyrs\cyrt\cyra \cyrs\cyrl\cyri\cyrk\cyra}%
+                    %{Листа слика}%
+  \def\listtablename{\CYRL\cyri\cyrs\cyrt\cyra \cyrt\cyra\cyrb\cyre\cyrl\cyra}%
+                    %{Листа табела}%
+  \def\indexname{\CYRI\cyrn\cyrd\cyre\cyrk\cyrs \cyrp\cyro\cyrje\cyrm\cyro\cyrv\cyra}%
+                %{Индекс појмова}%
+  \def\figurename{\CYRS\cyrl\cyri\cyrk\cyra}%{Слика}%
+  \def\tablename{\CYRT\cyra\cyrb\cyre\cyrl\cyra}%{Табела}%
+  \def\partname{\CYRD\cyre\cyro}%{Део}%
+  \def\enclname{\CYRP\cyrr\cyri\cyrl\cyro\cyrz\cyri}%{Прилози}%
+  \def\ccname{\CYRK\cyro\cyrp\cyri\cyrje\cyre}%{Копије}%
+  \def\headtoname{\CYRP\cyrr\cyri\cyrm\cyra}%{Прима}%
+  \def\pagename{\cyrs\cyrt\cyrr\cyra\cyrn\cyra}%{страна}%
+  \def\seename{\cyrv\cyri\cyrd\cyri}%{види}%
+  \def\alsoname{\cyrv\cyri\cyrd\cyri \cyrt\cyra\cyrk\cyro\cyrdje\cyre}%{види такође}%
+  \def\proofname{\CYRD\cyro\cyrk\cyra\cyrz}%{Доказ}%
+  \def\glossaryname{\CYRR\cyre\cyrch\cyrn\cyri\cyrk}%{Речник}%
+  }%
+%    \end{macrocode}
+%  \end{macro}
+%
+%  \begin{macro}{\dateserbianc}
+%    The macro |\dateserbianc| redefines the command |\today| to
+%    produce Serbian dates.
+%    \begin{macrocode}
+\def\dateserbianc{%
+  \def\today{\number\day .~\ifcase\month\or
+     \cyrje\cyra\cyrn\cyru\cyra\cyrr \or % јануар
+     \cyrf\cyre\cyrb\cyrr\cyru\cyra\cyrr \or % фебруар
+     \cyrm\cyra\cyrr\cyrt \or % март
+     \cyra\cyrp\cyrr\cyri\cyrl \or % април
+     \cyrm\cyra\cyrje \or % мај
+     \cyrje\cyru\cyrn \or % јун
+     \cyrje\cyru\cyrl \or % јул
+     \cyra\cyrv\cyrg\cyru\cyrs\cyrt \or % август
+     \cyrs\cyre\cyrp\cyrt\cyre\cyrm\cyrb\cyra\cyrr \or % септембар
+     \cyro\cyrk\cyrt\cyro\cyrb\cyra\cyrr \or % октобар
+     \cyrn\cyro\cyrv\cyre\cyrm\cyrb\cyra\cyrr \or % новембар
+     \cyrd\cyre\cyrc\cyre\cyrm\cyrb\cyra\cyrr\fi % децембар
+     \space \number\year.}}
+%    \end{macrocode}
+%  \end{macro}
+%
+%  \begin{macro}{\extrasserbianc}
+%  \begin{macro}{\noextrasserbianc}
+%    The macro |\extrasserbianc| will perform all the extra
+%    definitions needed for the Serbian language. The macro
+%    |\noextrasserbianc| is used to cancel the actions of
+%    |\extrasserbianc|.
+%
+%    In serbianc the first paragraph of each section should be indented.
+%    Add this code only in \LaTeX.
+%    \begin{macrocode}
+\ifx\fmtname plain \else
+  \let\@aifORI\@afterindentfalse
+  \def\bbl@frenchindent{\let\@afterindentfalse\@afterindenttrue
+                        \@afterindenttrue}
+  \def\bbl@nonfrenchindent{\let\@afterindentfalse\@aifORI
+                          \@afterindentfalse}
+  \addto\extrasserbianc{\bbl@frenchindent}
+  \addto\noextrasserbianc{\bbl@nonfrenchindent}
+\fi
+%    \end{macrocode}
+%  \end{macro}
+%  \end{macro}
+%
+%  \begin{macro}{\mathserbianc}
+%    Some math functions in serbianc math books have other names:
+%    e.g. |sinh| in serbianc is written as |sh| etc. So we define a
+%    number of new math operators.
+%    \begin{macrocode}
+\def\sh{\mathop{\operator@font sh}\nolimits} % same as \sinh
+\def\ch{\mathop{\operator@font ch}\nolimits} % same as \cosh
+\def\th{\mathop{\operator@font th}\nolimits} % same as \tanh
+\def\cth{\mathop{\operator@font cth}\nolimits} % same as \coth
+\def\arsh{\mathop{\operator@font arsh}\nolimits}
+\def\arch{\mathop{\operator@font arch}\nolimits}
+\def\arth{\mathop{\operator@font arth}\nolimits}
+\def\arcth{\mathop{\operator@font arcth}\nolimits}
+\def\tg{\mathop{\operator@font tg}\nolimits} % same as \tan
+\def\ctg{\mathop{\operator@font ctg}\nolimits} % same as \cot
+\def\arctg{\mathop{\operator@font arctg}\nolimits} % same as \arctan
+\def\arcctg{\mathop{\operator@font arcctg}\nolimits}
+\def\Prob{\mathop{\mathsf P\hskip0pt}\nolimits}
+\def\Expect{\mathop{\mathsf E\hskip0pt}\nolimits}
+\def\Variance{\mathop{\mathsf D\hskip0pt}\nolimits}
+%    \end{macrocode}
+%  \end{macro}
+%
+%    The macro |\ldf@finish| takes care of looking for a
+%    configuration file, setting the main language to be switched on
+%    at |\begin{document}| and resetting the category code of
+%    \texttt{@} to its original value.
+%    \begin{macrocode}
+\ldf@finish{serbianc}
+%</code>
+%    \end{macrocode}
+%
+% \Finale
+%% \CharacterTable
+%%  {Upper-case    \A\B\C\D\E\F\G\H\I\J\K\L\M\N\O\P\Q\R\S\T\U\V\W\X\Y\Z
+%%   Lower-case    \a\b\c\d\e\f\g\h\i\j\k\l\m\n\o\p\q\r\s\t\u\v\w\x\y\z
+%%   Digits        \0\1\2\3\4\5\6\7\8\9
+%%   Exclamation   \!     Double quote  \"     Hash (number) \#
+%%   Dollar        \$     Percent       \%     Ampersand     \&
+%%   Acute accent  \'     Left paren    \(     Right paren   \)
+%%   Asterisk      \*     Plus          \+     Comma         \,
+%%   Minus         \-     Point         \.     Solidus       \/
+%%   Colon         \:     Semicolon     \;     Less than     \<
+%%   Equals        \=     Greater than  \>     Question mark \?
+%%   Commercial at \@     Left bracket  \[     Backslash     \\
+%%   Right bracket \]     Circumflex    \^     Underscore    \_
+%%   Grave accent  \`     Left brace    \{     Vertical bar  \|
+%%   Right brace   \}     Tilde         \~}
+%%
+\endinput
Index: babel/serbianc.ins
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ babel/serbianc.ins	2011-06-06 13:07:21.775057330 +0200
@@ -0,0 +1,97 @@
+%%
+%% This file will generate fast loadable files and documentation
+%% driver files from the doc files in this package when run through
+%% LaTeX or TeX.
+%%
+%% Copyright 1989-2001 Johannes L. Braams and any individual authors
+%% listed elsewhere in this file.  All rights reserved.
+%% 
+%% This file is part of the Babel system, release 3.7.
+%% ---------------------------------------------------
+%% 
+%% It may be distributed under the terms of the LaTeX Project Public
+%% License, as described in lppl.txt in the base LaTeX distribution.
+%% Either version 1.2 or, at your option, any later version.
+%%
+%% --------------- start of docstrip commands ------------------
+%%
+\def\filedate{2011/06/06}
+\def\batchfile{serbianc.ins}
+\input docstrip.tex
+
+{\ifx\generate\undefined
+\Msg{**********************************************}
+\Msg{*}
+\Msg{* This installation requires docstrip}
+\Msg{* version 2.3c or later.}
+\Msg{*}
+\Msg{* An older version of docstrip has been input}
+\Msg{*}
+\Msg{**********************************************}
+\errhelp{Move or rename old docstrip.tex.}
+\errmessage{Old docstrip in input path}
+\batchmode
+\csname @@end\endcsname
+\fi}
+
+\declarepreamble\mainpreamble
+
+This is a generated file.
+
+Copyright 1989-2001 Johannes L. Braams and any individual authors
+listed elsewhere in this file.  All rights reserved.
+
+This file is part of the Babel system release 3.7.
+--------------------------------------------------
+
+It may be distributed under the terms of the LaTeX Project Public
+License, as described in lppl.txt in the base LaTeX distribution.
+Either version 1.2 or, at your option, any later version.
+\endpreamble
+
+\declarepreamble\fdpreamble
+This is a generated file.
+
+Copyright 2001 Johannes L. Braams and any individual authors
+listed elsewhere in this file.  All rights reserved.
+
+This file is part of the Babel system, release 3.7.
+---------------------------------------------------
+
+It may be distributed under the terms of the LaTeX Project Public
+License, as described in lppl.txt in the base LaTeX distribution.
+Either version 1.2 or, at your option, any later version.
+
+In particular, permission is granted to customize the declarations in
+this file to serve the needs of your installation.
+
+However, NO PERMISSION is granted to distribute a modified version
+of this file under its original name.
+
+\endpreamble
+
+\usedir{tex/generic/babel}
+\keepsilent
+ 
+\usepreamble\mainpreamble
+
+\generate{\file{serbianc.ldf}{\from{serbianc.dtx}{code}}}
+
+\usepreamble\fdpreamble
+ 
+\ifToplevel{
+\Msg{***********************************************************}
+\Msg{*}
+\Msg{* To finish the installation you have to move the following}
+\Msg{* files into a directory searched by TeX:}
+\Msg{*}
+\Msg{* \space\space All *.fd}
+\Msg{*}
+\Msg{* To produce the documentation run the files ending with}
+\Msg{* `.fdd' through LaTeX.}
+\Msg{*}
+\Msg{* Happy TeXing}
+\Msg{***********************************************************}
+}
+ 
+\endinput
